src/content/docs/en/core-concepts/routing.mdx

```diff
@@ -233,18 +233,36 @@ It's possible for multiple routes to match the same URL path. For example each o
 </FileTree>
 
 Astro needs to know which route should be used to build the page. To do so, it sorts them according to the following rules:
+1- Routes without path parameters
+- Endpoints without path parameters take precedence over all other routes
+- Routes without path parameters take precedence over dynamic routes
 
-- Static routes without path parameters will take precedence over all other routes
-- Dynamic routes using named parameters take precedence over rest parameters
+2- Dynamic routes with path parameters
+- Static endpoints using named parameters take precedence over other static routes
+- Static routes using named parameters take precedence over server endpoints using named parameters
+- Server endpoints using named parameters take precedence over server routes using named parameters
+- Server routes using named parameters take precedence over other routes using rest parameters
+
+3- Dynamic routes with rest parameters
 - Rest parameters have the lowest priority
-- Ties are resolved alphabetically
+- Static endpoints using rest parameters take precedence over other static routes using rest parameters
+- Static routes using rest parameters take precedence over server endpoints using rest parameters
+- Server endpoints using rest parameters take precedence over server routes using rest parameters
+- Server routes using rest parameters take precedence over other routes using rest parameters
```

Comment:
I wonder if it would be a good idea to create a [priority queue](https://www.google.com/search?q=Priority+Queue&tbm=isch) or a pyramid component to represent these